src/database/api/onDisconnect.ts

```diff
@@ -10,103 +10,104 @@ import {
 } from "../core/util/validation";
 import { warn } from "../core/util/util";
 import { Deferred } from "../../utils/promise";
+import { Repo } from '../core/Repo';
+import { Path } from '../core/util/Path';
 
 /**
  * @constructor
- * @param {!Repo} repo
- * @param {!Path} path
  */
-export const OnDisconnect = function(repo, path) {
-  /** @private */
-  this.repo_ = repo;
-
-  /** @private */
-  this.path_ = path;
-};
-
+export class OnDisconnect {
+  /**
+   * @param {!Repo} repo_
+   * @param {!Path} path_
+   */
+  constructor(private repo_: Repo,
+              private path_: Path) {
+  }
 
-/**
- * @param {function(?Error)=} opt_onComplete
- * @return {!firebase.Promise}
- */
-OnDisconnect.prototype.cancel = function(opt_onComplete) {
-  validateArgCount('Firebase.onDisconnect().cancel', 0, 1, arguments.length);
-  validateCallback('Firebase.onDisconnect().cancel', 1, opt_onComplete, true);
-  var deferred = new Deferred();
-  this.repo_.onDisconnectCancel(this.path_, deferred.wrapCallback(opt_onComplete));
-  return deferred.promise;
-};
+  /**
+   * @param {function(?Error)=} opt_onComplete
+   * @return {!firebase.Promise}
+   */
+  cancel(opt_onComplete) {
+    validateArgCount('firebase.database.onDisconnect().cancel', 0, 1, arguments.length);
```

Comment:
Sure. This originally said `Firebase.onDisconnect().cancel` so I changed the `Firebase` part to `firebase.database`. I've done the same in some other classes too, so just to be clear: in general should I remove the `Firebase`/`firebase.database` prefix and simply use the `ClassName.methodName` format?